### DISCUSSION TITLE
Manual download of maps and graphs

### DISCUSSION THREAD

**Discussion by ramondevic** [devemux86/cruiser#447] [Q&A] [answered]
I have two phones (I usually only use one but when I travel I bring another backup with all the apps and data) and I'm interested in downloading the GraphHopper maps and graphs from my home PC so I can test them from Cruiser Desktop and copy to the 2 phones.

Cruiser's automatic download is very good, but to do it manually, things get a little complicated.
I explain what I do because I'm sure there is a better and easier way to do it.

1) From Cruiser Desktop I download Mapsforge maps and GraphHopper graphs from here: https://download.mapsforge.org/
I copy them to a folder on my computer.
I start Cuiser Desktop, open the map and the graph.
I create a route and it works perfectly, I tried it with the new Spain graphs generated on 02-14-2024 (thanks devemux86 !)

2) I connect the phone to the PC by cable, transfer the files to the phone, open the map and perfect!
I do the same with the GraphHopper files, I copy the downloaded folder "Andorra" and rename it to "Andorra (manual)", I try to activate it by going to routing > graphhopper but it doesn't appear in the list.
I see that the other folders, e.g. "Spain (Mapsforge) have a text file named "Spain (Mapsforge)-cruiser.txt".
Create a file "Andorra (manual)-cruiser.txt" and change the contents of the "displayname" and "local.file" fields with "Andorra (manual)" and it appears, I can activate the graph and calculate a route.

My question is, could the app be modified to be able to work without these two txt files, just as Cruiser Desktop works?

Thanks !

**Comment by devemux86** [comment, marked as answer]
@ramondevic thanks for the question.

- **Offline maps**: enable the "Settings | Application | Advanced | Extended menu" and a new Map menu appears to open external offline map files.

- **GraphHopper graphs**: they do not support Android new storage restrictions, so we cannot open them externally.
On Android we must use Cruiser's download manager to get the offline GraphHopper graphs.

**Comment by devemux86** [comment]
> My question is, could the app be modified to be able to work without these two txt files, just as Cruiser Desktop works?

That's a good question and something I've thought about too, although Android is more complicated.
Anyway it's in my list to examine in the future. :slightly_smiling_face:

**Comment by MotoUKRider** [comment]
@ramondevic I've done a similar* thing with those text files and it's just a chore, so for my files I now do the reverse of your process: Download everything using the phone and then copy it to the PC as a backup and for Cruiser Desktop.  Once everything is on the PC I can then copy those files to my spare phone for testing.

*I was putting the Kurviger maps in place of the earlier Mapsforge ones until the rendering was tweaked so that city-names appeared sooner at the same zoom-level.

I think that makes us members of the 1% Club :smile:

**Reply by ramondevic** [reply]
A great idea this workaround, I hadn't thought of it.
I'm going to try it but I'm sure it will work perfectly with a lot less work.
Thanks !
